### PR TITLE
Make sure we always log in as app or provisioner

### PR DESCRIPTION
**Pull request by heaths**
This also adds previously-required parameters back into
@PSBoundParameters to pass down to pre- and post-scripts.

### CODE REVIEW

**Review by benbp — APPROVED** [review]

File: eng/common/TestResources/New-TestResources.ps1 [hunk 1]
```diff
@@ -177,13 +179,20 @@ if (!$CI -and !$TestApplicationId) {
...
+        # Make sure pre- and post-scripts are passed formerly required arguments.
+        $PSBoundParameters['TestApplicationId'] = $TestApplicationId
+        $PSBoundParameters['TestApplicationSecret'] = $TestApplicationSecret
 
-    Log "Created service principal '$TestApplicationId'"
+        Log "Created service principal '$TestApplicationId'"
+    }
 
     if (!$ProvisionerApplicationId) {
         $ProvisionerApplicationId = $TestApplicationId
```
Comment (pakrym) [inline eng/common/TestResources/New-TestResources.ps1:198]: We need to update PSBoundParameters here.

Comment (pakrym) [reply eng/common/TestResources/New-TestResources.ps1:198]: And ProvisionerApplicationSecret and TenantId

Comment (pakrym) [reply eng/common/TestResources/New-TestResources.ps1:198]: TestApplicationOid is also updated later in the script.

Comment (pakrym) [reply eng/common/TestResources/New-TestResources.ps1:198]: BaseName is updated later as well.

Comment (pakrym) [reply eng/common/TestResources/New-TestResources.ps1:198]: As well as ResourceGroupName

**Pull request merged**